src/ray/core_worker/core_worker.cc

```diff
@@ -1410,7 +1603,16 @@ void CoreWorker::HandleKillActor(const rpc::KillActorRequest &request,
     if (request.no_reconstruction()) {
       RAY_IGNORE_EXPR(local_raylet_client_->Disconnect());
     }
-    if (log_dir_ != "") {
+    if (options_.num_workers > 1) {
+      // TODO (kfstorm): Should we add some kind of check before sending the killing
+      // request?
+      RAY_LOG(ERROR)
+          << "Killing an actor which is running in a worker process with multiple "
+             "workers is not supported yet. The worker process will fail anyway, but you "
+             "should try to create the Java actor with some dynamic options to make it "
+             "being hosted in a dedicated worker process.";
```

Comment:
This comment is kind of misleading. By "not supported yet", I thought it means we don't support killing such an actor (the kill operation won't take effect). 
I think it's better to say "killing such an actor will also kill other actors in this process".

Reply:
Done.